src/main/java/io/pravega/connectors/flink/serialization/JsonRowDeserializationSchema.java

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.pravega.connectors.flink.serialization;
+
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.api.java.typeutils.RowTypeInfo;
+import org.apache.flink.streaming.util.serialization.DeserializationSchema;
+import org.apache.flink.types.Row;
+import org.apache.flink.util.Preconditions;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.io.IOException;
+
+/**
+ * Deserialization schema from JSON to {@link Row}.
+ *
+ * <p>Deserializes the <code>byte[]</code> messages as a JSON object and reads
+ * the specified fields.
+ *
+ * <p>Failure during deserialization are forwarded as wrapped IOExceptions.
+ */
+public class JsonRowDeserializationSchema implements DeserializationSchema<Row> {
+
+    /** Type information describing the result type. */
+    private final TypeInformation<Row> typeInfo;
+
+    /** Field names to parse. Indices match fieldTypes indices. */
+    private final String[] fieldNames;
+
+    /** Types to parse fields as. Indices match fieldNames indices. */
+    private final TypeInformation<?>[] fieldTypes;
+
+    /** Object mapper for parsing the JSON. */
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    /** Flag indicating whether to fail on a missing field. */
+    private boolean failOnMissingField;
+
+    /**
+     * Creates a JSON deserialization schema for the given fields and types.
+     *
+     * @param typeInfo   Type information describing the result type. The field names are used
+     *                   to parse the JSON file and so are the types.
+     */
+    public JsonRowDeserializationSchema(TypeInformation<Row> typeInfo) {
+        Preconditions.checkNotNull(typeInfo, "Type information");
+        this.typeInfo = typeInfo;
+
+        this.fieldNames = ((RowTypeInfo) typeInfo).getFieldNames();
+        this.fieldTypes = ((RowTypeInfo) typeInfo).getFieldTypes();
```

Comment:
Are we expecting `fieldNames` and `fileldTypes` size to match here?

Reply:
I'm emphasizing keeping this source file in sync with flink-kafka, and so don't plan to improve it.